tests/Unit/Evolution/DgSubcell/Test_Reconstruction.cpp

```diff
@@ -0,0 +1,239 @@
+        dg_det_jac, dg_mesh, subcell_mesh.extents());
+
+    // Our FD reconstruction scheme can integrate polynomials up to degree 6
+    // exactly. However, we want to verify that if we have more than 8 grid
+    // points on the DG grid that we still are able to recover the correct
+    // solution.
+    const DataVector expected_nodal_coeffs =
+        TestHelpers::evolution::dg::subcell::cell_values(
+            std::min(dg_mesh.extents(0) - 2, 6_st), inertial_coords);
+
+    const DataVector subcell_values =
+        TestHelpers::evolution::dg::subcell::cell_values(
+            std::min(dg_mesh.extents(0) - 2, 6_st), subcell_inertial_coords);
+
+    const DataVector reconstructed_datavector =
+        evolution::dg::subcell::fd::reconstruct(
+            subcell_values * projected_det_jac, dg_mesh,
+            subcell_mesh.extents()) /
+        dg_det_jac;
+
+    // Test reconstruction of a DataVector
+    CHECK_ITERABLE_CUSTOM_APPROX(reconstructed_datavector,
+                                 expected_nodal_coeffs, local_approx);
+
+    // Test reconstruction of a Variables
+    Variables<tmpl::list<Tags::Scalar, Tags::Vector<Dim>>> cell_values_vars(
+        subcell_mesh.number_of_grid_points());
+    get(get<Tags::Scalar>(cell_values_vars)) =
+        TestHelpers::evolution::dg::subcell::cell_values(
+            std::min(dg_mesh.extents(0) - 2, 6_st), subcell_inertial_coords) *
+        projected_det_jac;
+    for (size_t d = 0; d < Dim; ++d) {
+      get<Tags::Vector<Dim>>(cell_values_vars).get(d) =
+          (d + 2.0) *
+          TestHelpers::evolution::dg::subcell::cell_values(
+              std::min(dg_mesh.extents(0) - 2, 6_st), subcell_inertial_coords) *
+          projected_det_jac;
+    }
+
+    const auto check_each_field_in_vars = [&dg_det_jac, &expected_nodal_coeffs,
+                                           &local_approx](
+                                              const auto& local_dg_vars) {
+      CHECK_ITERABLE_CUSTOM_APPROX(
+          DataVector(get(get<tmpl::front<typename std::decay_t<decltype(
+                             local_dg_vars)>::tags_list>>(local_dg_vars)) /
+                     dg_det_jac),
+          expected_nodal_coeffs, local_approx);
+      for (size_t d = 0; d < Dim; ++d) {
+        CHECK_ITERABLE_CUSTOM_APPROX(
+            DataVector(get<tmpl::back<typename std::decay_t<decltype(
+                           local_dg_vars)>::tags_list>>(local_dg_vars)
+                           .get(d) /
+                       dg_det_jac),
+            (d + 2.0) * expected_nodal_coeffs, local_approx);
+      }
+    };
+
+    Variables<tmpl::list<Tags::Scalar, Tags::Vector<Dim>>>
+        reconstructed_vars = evolution::dg::subcell::fd::reconstruct(
+            cell_values_vars, dg_mesh, subcell_mesh.extents());
+    check_each_field_in_vars(reconstructed_vars);
+
+    // Check not_null with no prefix tags
+    reconstructed_vars.initialize(0);
+    check_each_field_in_vars(reconstructed_vars);
```

Comment:
Please clarify the "Check not_null" comment above. As far as I can tell you're passing in an empty vars and comparing it against non-empty datavectors...?

Reply:
I don't understand how this passed the test, clearly some check in for sizes somewhere is lost... I added a `REQUIRE` to ensure num points matches up, then this fails.